treeherder/etl/perf_data_adapters.py

```diff
@@ -90,16 +90,19 @@ def _round(num):
         return round(num, 2)
 
     @staticmethod
-    def _calculate_summary_data(job_id, result_set_id, push_timestamp, results):
+    def _calculate_summary_data(job_id, result_set_id, push_timestamp,
+                                results, summary):
```

Comment:
I don't passing a summary to this method is quite right -- I think we should just use the summary inline if it's available, and keep this method focused on calculating the summary where needed.